Sources/TuistSupport/Utils/SignPosts.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+import os.log
+
+public class Signpost {
+    private let identifier: StaticString
+    private let label: String?
+    private let log: OSLog
+    private let signpostID: OSSignpostID
+
+    public init(category: StaticString, identifier: StaticString, label: String? = nil) {
+        self.identifier = identifier
+        self.label = label
+        log = OSLog(subsystem: "io.tuist", category: "\(category)")
+        signpostID = OSSignpostID(log: log)
+    }
+
+    public func begin() {
+        Signpost.createSignPost(.begin, log: log, name: identifier, signpostID: signpostID, label: label)
+    }
+
+    public func end(label: String? = nil) {
+        Signpost.createSignPost(.end, log: log, name: identifier, signpostID: signpostID, label: label ?? self.label)
+    }
+
+    public static func measure<T>(
+        category: StaticString,
+        identifier: StaticString,
+        label: String? = nil,
+        _ code: () throws -> T
```

Comment:
Not sure about the testability of making this a static function this way. 
I'm not sure _if_ you even plan to test it but you might want to test the order of execution or things like that.

Reply:
Good point, ya would be good to be able to have an alternative implementation too _(e.g. one that prints the results to the command line)_ 

May have to have a higher level type to help orchestrate this e.g. `SignpostsFactory` which would have instance methods:

- `measure(...)`: the convenience for automatically measuring a code block
- `makeSignpost() -> Signpost`: for manual `.begin()` and `.end()` measurements